### PR TITLE
Add --skip_push option to setup

### PR DESCRIPTION
**Pull request by GeNiuS69**
Add possibility to skip build and push image on setup if you already have builded image in registry, e.g. after GitLab CI/CD.

### CODE REVIEW

**Comment by igor-alexandrov** [comment]
@GeNiuS69 you passed for no reason `invoke_options` to `kamal:cli:server:bootstrap`, `kamal:cli:env:push` and to `kamal:cli:accessory:boot`. However the only change that is really need to make this work is the change you made on line 3: https://github.com/basecamp/kamal/pull/558/files#diff-cd863d3bebefa5ecfbacd577abdff981aa7695ff9c0a39a634ac1baaee3901b2R3.

Please update the PR accordingly.

**Comment by GeNiuS69** [comment]
@igor-alexandrov done

File: test/cli/main_test.rb [hunk 1]
```diff
@@ -2,12 +2,17 @@
 
 class CliMainTest < CliTestCase
   test "setup" do
+    invoke_options = { "config_file" => "test/fixtures/deploy_simple.yml", "version" => "999", "skip_hooks" => false }
+
```
Comment (Sija) [inline test/cli/main_test.rb:6]: That seems unneeded as well:

```suggestion
```

Comment (igor-alexandrov) [reply test/cli/main_test.rb:6]: Yes, but at the same time I don't see tests, which cover `skip_push` option for the `setup` task.

Comment (GeNiuS69) [reply test/cli/main_test.rb]: Yes, sure, my bad. Removed

File: lib/kamal/cli/main.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class Kamal::Cli::Main < Kamal::Cli::Base
   desc "setup", "Setup all accessories, push the env, and deploy app to servers"
+  option :skip_push, aliases: "-P", type: :boolean, default: false, desc: "Skip image build and push"
```
Comment (igor-alexandrov) [inline lib/kamal/cli/main.rb:3]: Please add tests for the `skip_push` option similar to https://github.com/basecamp/kamal/blob/main/test/cli/main_test.rb#L40

**Review by igor-alexandrov — CHANGES_REQUESTED** [review]

File: lib/kamal/cli/main.rb (hunk 2 above)
Comment (kalashnikovisme) [reply lib/kamal/cli/main.rb:3]: Looks like tests were added :slightly_smiling_face: 

Is there something that prevents us from having this wonderful feature in the release? :slightly_smiling_face:

**Comment by kalashnikovisme** [comment]
Tried this version of gem locally.

```
~: kamal setup --skip-push
  INFO [e011c4ce] Running /usr/bin/env mkdir -p .kamal on 165.227.149.148
  INFO [e011c4ce] Finished in 1.337 seconds with exit status 0 (successful).
Acquiring the deploy lock...
Ensure Docker is installed...
Releasing the deploy lock...
  Finished all in 1.9 seconds
ERROR: "kamal server bootstrap" was called with arguments ["--skip-push"]
Usage: "kamal server bootstrap"
```

Trying to figure out what is going on :slightly_smiling_face:

**Comment by GeNiuS69** [comment]
@kalashnikovisme hmm, it was my initial solution, but I couldn't defend it

**Comment by igor-alexandrov** [comment]
@GeNiuS69 sorry, it was my fault, now I understood why you used `deploy_options` to pass params to internal tasks.
@kalashnikovisme in your solution assignment of `invoke_options = deploy_options` seems to be unnecessary, `deploy_options` can be used directly.

**Comment by kalashnikovisme** [comment]
@igor-alexandrov yeah! You're right :slightly_smiling_face: I don't like this assignment `invoke_options = deploy_options` either. I've added because the same was in the next function 

```ruby
def deploy
    runtime = print_runtime do
      mutating do
        invoke_options = deploy_options # <-----------------------
        # ...
      end
    end
end
```

**Comment by djmb** [comment]
Thanks for the PR @GeNiuS69!

**Comment by zmoon460** [comment]
Very important and useful feature, thank you

**Pull request merged**